### PR TITLE
Implement putValue and getValue Unsafe APIs

### PR DESCRIPTION
**Pull request by ehrenjulzert**
for #13614
Signed-off-by: Ehren Julien-Neitzert <ehren.julien-neitzert@ibm.com>

### CODE REVIEW

File: runtime/vm/BytecodeInterpreter.hpp [hunk 1]
```diff
@@ -3981,24 +3981,75 @@ class INTERPRETER_CLASS
...
-		return EXECUTE_BYTECODE;
+		j9object_t clz = *(j9object_t*)_sp;
+		I_64 offset = *(I_64*)(_sp + 1);
+		j9object_t obj = *(j9object_t*)(_sp + 3);
+
+		j9object_t result = NULL;
+		VM_BytecodeAction rc = EXECUTE_BYTECODE;
+
+		/* TODO (#14073): update this function to have the same behavior as OpenJDK when obj is null, clz is null, or when clz is not a VT class (currently OpenJDK segfaults in all of these scenarios) */
+		if (NULL != obj && NULL != clz) {
```
Comment (tajila) [inline runtime/vm/BytecodeInterpreter.hpp:3996]: You are missing the fast path allocation here.

Comment (tajila) [reply runtime/vm/BytecodeInterpreter.hpp:3996]: I think a simple way to handle this is to take parts out of  the `ValueTypeHelpers::getFlattenableField` function.

this part
```
			if (fastPath) {
				returnObjectRef = objectAllocate.inlineAllocateObject(currentThread, flattenedFieldClass, false, false);
				if (NULL == returnObjectRef) {
					goto done;
				}
			} else {
				VM_VMHelpers::pushObjectInSpecialFrame(currentThread, receiver);
				returnObjectRef = vm->memoryManagerFunctions->J9AllocateObject(currentThread, flattenedFieldClass, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
				receiver = VM_VMHelpers::popObjectInSpecialFrame(currentThread);
				if (J9_UNEXPECTED(NULL == returnObjectRef)) {
					goto done;
				}
				flattenedFieldClass = VM_VMHelpers::currentClass(flattenedFieldClass);
			}

			if (J9CLASS_HAS_4BYTE_PREPADDING(flattenedFieldClass)) {
				returnObjectOffset += sizeof(U_32);
			}

			objectAccessBarrier.copyObjectFields(currentThread,
								flattenedFieldClass,
								receiver,
								cache->offset + objectHeaderSize,
								returnObjectRef,
								objectHeaderSize + returnObjectOffset);
```

and make it a separate helper function and just call that here. That way you dont need to duplicate code.

File: runtime/vm/BytecodeInterpreter.hpp [hunk 2]
```diff
@@ -3981,24 +3981,75 @@ class INTERPRETER_CLASS
...
+		j9object_t obj = *(j9object_t*)(_sp + 4);
+
+		VM_BytecodeAction rc = EXECUTE_BYTECODE;
+
+		/* TODO (#14073): update this function to have the same behavior as OpenJDK when obj is null, clz is null, or when clz is not a VT class (currently OpenJDK segfaults in all of these scenarios) */
+		if ((NULL != obj) && (NULL != clz) && (NULL != value)) {
+			J9Class *clzJ9Class = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, clz);
+
+			if (J9_IS_J9CLASS_VALUETYPE(clzJ9Class)) {
+				UDATA clzDataStart = (UDATA)J9VMTHREAD_OBJECT_HEADER_SIZE(_currentThread) + J9CLASS_PREPADDING_SIZE(clzJ9Class);
```
Comment (tajila) [inline runtime/vm/BytecodeInterpreter.hpp:4039]: same here

File: runtime/vm/BytecodeInterpreter.hpp (hunk 1 above)
Comment (ehrenjulzert) [reply runtime/vm/BytecodeInterpreter.hpp:3996]: Okay, I'll get working on this now

Comment (ehrenjulzert) [reply runtime/vm/BytecodeInterpreter.hpp:3996]: addressed in bcc3d7b

File: runtime/vm/BytecodeInterpreter.hpp (hunk 2 above)
Comment (ehrenjulzert) [reply runtime/vm/BytecodeInterpreter.hpp]: addressed in bcc3d7b

**Comment by tajila** [comment]
Jenkins test sanity,extended plinuxval jdknext

**Comment by tajila** [comment]
Jenkins test sanity zlinuxvalst jdknext

**Comment by tajila** [comment]
Jenkins test sanity win jdk8

**Comment by tajila** [comment]
Jenkins test sanity plinuxvalst jdknext

**Comment by tajila** [comment]
crash on s390 is a known issue

**Pull request merged**